### PR TITLE
Try to get this compiling with nix

### PR DESCRIPTION
**Pull request by cdepillabout**
With the changes in this PR, I've got something that will at least start compiling with `nix-build`.

### CODE REVIEW

File: default.nix [hunk 1]
```diff
@@ -1,10 +1,52 @@
...
+    } {};
+
+    # There are various dependencies that appear to be marked as broken.  Here,
+    # I manually mark them as unbroken.  Let's hope that they do actually
+    # compile.
+
+    aeson-extra = haskell-lib.overrideCabal prev.aeson-extra { broken = false; };
+    hedgehog-fn = haskell-lib.overrideCabal prev.hedgehog-fn { broken = false; };
+    time-parsers = haskell-lib.overrideCabal prev.time-parsers { broken = false; };
+    nonempty-containers = haskell-lib.overrideCabal prev.nonempty-containers { broken = false; };
```
Comment (cdepillabout) [inline default.nix]: I've started running `nix-build`, but my big worry is whether or not these libraries are _actually_ broken, or they are just incorrectly marked as broken.

If they are incorrectly marked as broken, then just manually marking them as non-broken should be fine.

If they are _actually_ broken, then there may be some things that need to happen to get them to start working.  Possibly bump them up to a later version, disable failing tests, etc.

**Comment by cdepillabout** [comment]
The previous two commits get everything building (well, except for `app`, which has a problem in it).

You may also be interested in https://discourse.nixos.org/t/nix-haskell-development-2020/6170, which gives some tips for Haskell development with nix.

File: default.nix [hunk 2]
```diff
@@ -1,10 +1,76 @@
...
+    scientific = haskell-lib.dontCheck prev.scientific;
+    base-compat-batteries = haskell-lib.dontCheck prev.base-compat-batteries;
+
+    # A bunch of things use doctest in their tests, but doctest doesn't appear
+    # to be able to be compiled, so we just disable tests for all these.
+    comonad = haskell-lib.dontCheck prev.comonad;
+    lens = haskell-lib.dontCheck prev.lens;
+    semigroupoids = haskell-lib.dontCheck prev.semigroupoids;
+    http-types = haskell-lib.dontCheck prev.http-types;
+    servant = haskell-lib.dontCheck prev.servant;
```
Comment (cdepillabout) [inline default.nix:65]: It seems strange that the haskell package set that `miso` provides doesn't have these types of packages already fixed up.

It seems like something is probably setup incorrectly here.

For instance, it is hard to believe that `miso` doesn't have `QuickCheck` in it's transitive dependency list.

**Comment by ChrisPenner** [comment]
This got it building! Close to having a real web-app now 👌 

Thanks so much ❤️

**Pull request merged**